endpoints/cookie_sync.go

```diff
@@ -122,7 +122,7 @@ func (c *cookieSyncEndpoint) parseRequest(r *http.Request) (usersync.Request, ma
 
 	request := cookieSyncRequest{}
 	if err := jsonutil.UnmarshalValid(body, &request); err != nil {
-		return usersync.Request{}, macros.UserSyncPrivacy{}, fmt.Errorf("JSON parsing failed: %s", err.Error())
+		return usersync.Request{}, macros.UserSyncPrivacy{}, fmt.Errorf("request JSON is invalid")
```

Comment:
While I agree it's better not to directly return the error message, a generalized error does not provide any information about what was invalid. Thoughts?

Reply:
Would "Malformed JSON" be more intuitive? The user should be able to validate the json using one of the online or command-line verifiers, which can provide more specific instructions. Would that suffice?

Reply:
My concern is the request json is ephemeral and may not be available, whereas the error is more likely to be logged. 

Let's take a step back, what issue are you targeting to fix with this change?

Reply:
> what issue are you targeting to fix with this change?

To not expose implementation specific errors to the client, as a security best practice. Logging the specific error message and/or capturing metrics around it seems more reasonable.